docs/migrations/migrations_guide.md

```diff
@@ -0,0 +1,192 @@
+
+On application startup, the code in `migration.go` runs all unapplied `.up.sql` files in order, using `golang-migrate`'s `Up()` function.
+
+**Automatic Execution:**
+- Migrations run when the app starts.
+- Already-applied migrations are skipped.
+- Locking ensures only one process can apply them at a time (via advisory locks).
+
+> ✅ No manual steps are needed if your migration files are correctly placed and committed.
+
+---
+
+## ✏️ Creating a New Migration
+
+Use the CLI to generate a new migration file pair:
+
+```bash
+migrate create -ext sql -dir ./migrations -seq add_new_column_to_users
+```
+
+This creates:
+
+```
+000002_add_new_column_to_users.up.sql
+000002_add_new_column_to_users.down.sql
+```
+
+### ✍️ Edit the files:
+
+**Up (apply):**
+
+```sql
+ALTER TABLE users ADD COLUMN nickname TEXT;
+```
+
+**Down (rollback):**
+
+```sql
+ALTER TABLE users DROP COLUMN nickname;
+```
+
+---
+
+## 🕒 When Migrations Run
+
+- Automatically on service startup.
+- On local dev, you can trigger them manually via CLI (see below).
+- Only `.up.sql` files are executed in forward direction, and only if not already applied.
+
+---
+
+## 🔁 Rollbacks
+
+To undo one or more migrations:
+
+### Using CLI:
+
+```bash
+migrate -path ./migrations -database <DB_URL> down 1     # Rollback last 1 migration
+migrate -path ./migrations -database <DB_URL> down        # Rollback all
+```
+
+### Programmatically:
+
+In Go:
+
+```go
+_ = m.Steps(-1)   // Rollback one
+_ = m.Down()      // Rollback all
+```
+
+### Dirty state recovery:
+
+If a migration fails midway:
+- The DB may be marked “dirty”
+- Use `migrate force <version>` to fix state before rerunning
+
+---
+
+## 🧪 How to Test Migrations
+
+1. Spin up a local DB (Docker or Postgres locally)
+2. Run all migrations up:
+   ```bash
+   migrate -path ./migrations -database <DB_URL> up
+   ```
+3. Run all down:
+   ```bash
+   migrate -path ./migrations -database <DB_URL> down
+   ```
+4. Run tests (if defined) to verify schema is as expected
+
+---
+
+## 🧾 Naming Conventions
+
+- Use `snake_case` for clarity
+- Be descriptive: `add_index_to_org_table`, `create_user_settings_table`, etc.
+- Keep versions sequential (`000001`, `000002`, ...) or use timestamps (`20250615120000_create_table.up.sql`)
+
+---
+
+## 🛠️ Best Practices
+
+- Keep migrations **small and focused**
+- Always add both `.up.sql` and `.down.sql`
+- **Test migrations** in a clean DB locally before pushing
+- Do **not edit** already-applied migration files—create a new one
+- Include migration tests in CI/CD
+- Prefer **embedded migrations** in binary for consistent deploys
+
+---
+
+## 📎 Helpful Commands
+
+```bash
+# Install CLI
+brew install golang-migrate
+
+# Run migrations
+migrate -path ./migrations -database postgres://user:pass@localhost:5432/db up
+
+# Rollback
+migrate -path ./migrations -database postgres://user:pass@localhost:5432/db down 1
+
+# Fix dirty state
+migrate -path ./migrations -database <DB_URL> force <version>
+```
```

Comment:
We have Bun ORM installed as part of project dependency hence will be added as part of the dev setup.

Current document says to use `golang-migrate`, instead we should use bun ORM which is already part of the project

Reply:
The system recursively scans the `api/migrations/` directory automatically runs the scripts

Something like this: 

```text
Application Start > .env gets populated > database connection 
> migration system init > discover migration files > check applied migrations 
> Execute Pending Migrations > Update Migration Tracking > Continue Application Startup
```

Right now we do not have any dedicated/independent way to run the script as such. 

@raghavyuva: Correct me if i am wrong the last point